### DISCUSSION TITLE
NuGet for HAST

### DISCUSSION THREAD

**Discussion by Evanion** [unifiedjs/unified#239] [Q&A]
I was wondering if anyone know of any NuGet package that will produce HAST compatible trees?
Our projects backend is in c# and we would like to be able to use HAST as a contract between the backend and frontend.

**Comment by ChristianMurphy** [comment]
Welcome @Evanion! 👋 

> Our projects backend is in c# and we would like to be able to use HAST as a contract between the backend and frontend.

Taking a step back, you probably don't want to do that.
AST really aren't meant as a wire format https://github.com/orgs/unifiedjs/discussions/188#discussioncomment-2466114
It is neither more efficient nor more secure than just sending HTML.
Before going down the road of solving the [XY Problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem) of "how do I generate a HAST tree in C#?", take a moment to step back and answer the question: "why do I want to do that?".

---

Answering your direct question 

> I was wondering if anyone know of any NuGet package that will produce HAST compatible trees?

there isn't one, but you could build a wrapper around an existing parser like https://github.com/AngleSharp/AngleSharp to generate one.

**Reply by Evanion** [reply]
The reason we want an AST rather than sending pure HTML, is that we want to render platform native elements in different platforms. for instance in web and native apps.
This includes more complex elements than regular html have (like widgets).

**Reply by wooorm** [reply]
HTML has custom elements now, I’d think that you could use that for those widgets

**Reply by Evanion** [reply]
Yea I know about web components, but we want to use native react components.

**Reply by wooorm** [reply]
my point is that you can use HTML as an over-the-wire format and map web components to anything
